### PR TITLE
feat(tekton): support official bundle resolver

### PR DESCRIPTION
**Pull request by lcarva**
## Changes

The functionality provided by the tektoncd/resolution project has been added directly to the main tektoncd/pipelines project. Thus, tektoncd/resolution is now deprecated.

As part of this move, one of the attributes in the CRD change names from "resource" to "params". This commit drops support for the deprecated "resource" attribute, and adds support for the officially supported "params" attribute.

The docs have also been updated to reference the official documentation instead of the dreprecated GitHub repository.

## Context

See above.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/tekton/extract.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ function addDep(ref: TektonBundle, deps: PackageDependency[]): void {
   let imageRef: string | undefined;
   // Find a bundle reference from the Bundle resolver
   if (ref.resolver === 'bundles') {
-    for (const field of coerceArray(ref.resource)) {
+    for (const field of coerceArray(ref.params)) {
```
Comment (viceice) [inline lib/modules/manager/tekton/extract.ts:72]: we should support the deprecated resources and provide a migration instead.

a lot of our managers support deprecated features.

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/tekton/extract.ts (hunk 1 above)
Comment (lcarva) [reply lib/modules/manager/tekton/extract.ts]: ack, makes sense. done

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/tekton/types.ts [hunk 2]
```diff
@@ -17,10 +17,11 @@ interface TektonResourceSpec {
 export interface TektonBundle {
   bundle: string;
   resolver: string;
-  resource: TektonBundleResourceField[];
+  resource: TektonResolverParamsField[];
+  params: TektonResolverParamsField[];
```
Comment (viceice) [inline lib/modules/manager/tekton/types.ts:21]:
```suggestion
  resource?: TektonResolverParamsField[];
  params?: TektonResolverParamsField[];
```

those are nullable

File: lib/modules/manager/tekton/extract.spec.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ describe('modules/manager/tekton/extract', () => {
         'test-file.yaml'
       );
       expect(result).toMatchSnapshot();
-      expect(result?.deps).toHaveLength(16);
+      expect(result?.deps).toHaveLength(27);
```
Comment (viceice) [inline lib/modules/manager/tekton/extract.spec.ts:12]: Do we really need so many new fixtures? Can we remove most of the old `resource` fixtures and only leave one as they are the same as the new `params` beside the property rename?

File: lib/modules/manager/tekton/readme.md [hunk 4]
```diff
@@ -13,16 +13,17 @@ Tasks and Pipeline definitions can also live outside the Kubernetes cluster and
 The `tekton` manager focuses on providing updates to Tekton resource references.
 
 Right now, Renovate's Tekton manager only supports references that are [Bundles](https://tekton.dev/docs/pipelines/tekton-bundle-contracts/).
-See the [`tektoncd/resolution` project on GitHub](https://github.com/tektoncd/resolution) for the different kinds of Tekton references.
+See the [docs](https://tekton.dev/docs/pipelines/resolution/) for the different kinds of Tekton references and their corresponding resolvers.
```
Comment (HonkingGoose) [inline lib/modules/manager/tekton/readme.md:16]:
```suggestion
Read the [Tekton Pipeline remote resolution docs](https://tekton.dev/docs/pipelines/resolution/) for the different kinds of Tekton references and their corresponding resolvers.
```

Let's use a clear link name.

File: lib/modules/manager/tekton/readme.md [hunk 5]
```diff
@@ -13,16 +13,17 @@ Tasks and Pipeline definitions can also live outside the Kubernetes cluster and
...
 ### Using a Tekton Bundle reference
 
-There are two ways to use a Tekton Bundle reference:
+There are three ways to use a Tekton Bundle reference:
 
-1. Via the [`tektoncd/resolution` project](https://github.com/tektoncd/resolution)
-2. Via the `taskRun.spec.taskRef.bundle` and the `pipelineRun.spec.pipelineRef.bundle` attributes
+1. Via the [Bundles Resolver](https://tekton.dev/docs/pipelines/bundle-resolver/)
+2. Via the [`tektoncd/resolution` project](https://github.com/tektoncd/resolution)
+3. Via the `taskRun.spec.taskRef.bundle` and the `pipelineRun.spec.pipelineRef.bundle` attributes
```
Comment (HonkingGoose) [inline lib/modules/manager/tekton/readme.md:24]:
```suggestion
1. Via the [Tekton Bundles Resolver](https://tekton.dev/docs/pipelines/bundle-resolver/)
1. Via the [`tektoncd/resolution` project](https://github.com/tektoncd/resolution)
1. Via the `taskRun.spec.taskRef.bundle` and the `pipelineRun.spec.pipelineRef.bundle` attributes
```

We can just use `1.` and let the Markdown renderer order it for us. That way we can't get the numbering wrong. 😉

And we can use a better link name.

File: lib/modules/manager/tekton/readme.md [hunk 6]
```diff
@@ -13,16 +13,17 @@ Tasks and Pipeline definitions can also live outside the Kubernetes cluster and
...
 
-Renovate's Tekton manager supports both methods.
+Renovate's Tekton manager supports all of them.
```
Comment (HonkingGoose) [inline lib/modules/manager/tekton/readme.md:26]:
```suggestion
Renovate's Tekton manager supports all the methods listed above.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Some small docs changes.

File: lib/modules/manager/tekton/readme.md [hunk 7]
```diff
@@ -13,16 +13,17 @@ Tasks and Pipeline definitions can also live outside the Kubernetes cluster and
 The `tekton` manager focuses on providing updates to Tekton resource references.
 
 Right now, Renovate's Tekton manager only supports references that are [Bundles](https://tekton.dev/docs/pipelines/tekton-bundle-contracts/).
-See the [`tektoncd/resolution` project on GitHub](https://github.com/tektoncd/resolution) for the different kinds of Tekton references.
+Read the [Tekton Pipeline remote resolution docs](https://tekton.dev/docs/pipelines/resolution/) for the different kinds of Tekton references and their corresponding resolvers.
 
 ### Using a Tekton Bundle reference
 
-There are two ways to use a Tekton Bundle reference:
+There are three ways to use a Tekton Bundle reference:
```
Comment (viceice) [inline lib/modules/manager/tekton/readme.md:20]: is this still true? i only see two list items

Comment (lcarva) [reply lib/modules/manager/tekton/readme.md:20]: It should be:

1. Via the [Tekton Bundles Resolver](https://tekton.dev/docs/pipelines/bundle-resolver/)
1. Via the [`tektoncd/resolution` project](https://github.com/tektoncd/resolution)
1. Via the `taskRun.spec.taskRef.bundle` and the `pipelineRun.spec.pipelineRef.bundle` attributes

I believe that's what the diff says.

Comment (viceice) [reply lib/modules/manager/tekton/readme.md:20]: i don't know. i only see that it needs a third list items or revert to two. 🤷‍♂️😉

Comment (lcarva) [reply lib/modules/manager/tekton/readme.md:20]: Maybe I'm missing something... Here's what the rendered markdown file looks like:
image

**Review by viceice — APPROVED** [review]

**Review by HonkingGoose — APPROVED** [review]

**Pull request merged**